### PR TITLE
Frontend Homework 5

### PR DESCRIPTION
**Pull request by asish918**
## Frontend Homework 5
I know we are not supposed to commit the .env file to github. I did this on purpose so that you can see the file and its contents in my PR.

## Drift Video
[Drift Video](https://video.drift.com/v/abwDqxkwxuhaM3nxxn9cKurnl5mt8vD8YsqCSOKODg4ro)

## Screenshots

Q1 OS Details GET Reponse
Screenshot 2024-02-12 at 22 22 26

Q2
Screenshot 2024-02-12 at 22 22 00

index.html page served from the server with CSS
Screenshot 2024-02-12 at 22 21 37

### CODE REVIEW

**Comment by asish918** [comment]
| Metric | Value |
| ------ | ----- |
| alert_status | <b>:white_check_mark: OK</b> |
| bugs | 0 |
| code_smells | 0 |
| reliability_rating | A (0 Bugs)<br/>Note: A being best, E being worst |
| security_rating | A (0 Vulnerabilities)<br/>Note: A being best, E being worst |
| vulnerabilities | 0 |

**Comment by Nitesh2503** [comment]
Always add proper commit messages
Follow this message syntax `<branch_name> : "Commit message (a bit descriptive)">`
And if too many commits are there then squash them

File: frontend/homework/5/index.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
```
Comment (Nitesh2503) [inline frontend/homework/5/index.html:1]: The details format must be according to the provided format

File: frontend/homework/5/os_info.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        os_release: os.platform(),
+        uptime: os.uptime(),
+        core_count: os.cpus().length,
+        total_memory: os.totalmem(),
+        free_memory: os.freemem(),
+    }
+
+
+    fs.writeFileSync(path.join(__dirname + "/obj.json"), JSON.stringify(os_json), (err) => {
+        if (err) throw new err;
```
Comment (Nitesh2503) [inline frontend/homework/5/os_info.js:19]: Use `{}` always whether it's a single line or not

File: frontend/homework/5/server.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+const http = require('http');
+
+const os_info = require('./os_info');
+const q2 = require('./q2');
+
+require('dotenv').config();
+
+const server = http.createServer((req, res) => {
+    switch (req.url) {
+        case '/':
```
Comment (Nitesh2503) [inline frontend/homework/5/server.js:12]: Good implementation via using the switch case

File: frontend/homework/5/server.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            fs.readFile('obj.json', (err, data) => {
+                if (err) {
+                    res.writeHead(500, { 'Content-Type': 'text/plain' });
+                    res.end('Internal Server Error');
+                } else {
+                    res.writeHead(200, { 'Content-Type': 'application/json' });
```
Comment (Nitesh2503) [inline frontend/homework/5/server.js:18]: Missed some data that also needs to be shown, go through the doc

File: frontend/homework/5/style.css [hunk 5]
```diff
@@ -0,0 +1,112 @@
+html {
```
Comment (Nitesh2503) [inline frontend/homework/5/style.css:1]: Use sass

File: frontend/homework/5/server.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+                    res.end(data);
+                }
+            });
+            break;
+
+        case '/index.html':
+            fs.readFile('obj.json', 'utf-8', (err, data) => {
```
Comment (Nitesh2503) [inline frontend/homework/5/server.js:25]: Could be separated into functions

**Pull request closed without merging**